optimade/server/schemas.py

```diff
@@ -0,0 +1,126 @@
+from optimade.models import (
+    StructureResource,
+    ReferenceResource,
+    EntryResource,
+    DataType,
+)
+from .config import CONFIG
+from typing import Dict, Union
+
+
+def retrieve_queryable_properties(
+    schema: dict,
+    queryable_properties: list,
+    entry_provider_fields: list = None,
+    base_schema: dict = None,
+) -> dict:
+    """For a given schema dictionary and a list of desired properties,
+    recursively descend into the schema and set the appropriate values
+    for the `description`, `sortable` and `unit` keys, returning the expanded
+    schema dict that includes nested schemas.
+
+    """
+    properties = {}
+    if entry_provider_fields is None:
+        entry_provider_fields = []
+
+    if base_schema is None:
+        base_schema = schema.copy()
+
+    for name, value in schema["properties"].items():
+        if name in queryable_properties:
+            _keys = ("allOf", "anyOf", "oneOf")
+            for key in _keys:
+                if key in value:
+                    condOf = value.pop(key)
+                    if isinstance(condOf, list):
+                        value.update(condOf[0])
```

Comment:
Should just comment here that this was a bit of late-night hack; we need to think more about how to unwrap the case where multiple types are allowed for a given field. Here I just grab the first, for the sake of testing.